src/backend/distributed/README.md

```diff
@@ -0,0 +1,2445 @@
+
+**Worker nodes**: Nodes with groupid > 0, which can perform reads and writes, but not administrative operations.
+
+**Cluster**: The combination of worker nodes and coordinator is a cluster. When the cluster has a read replica, the nodes in the read replica are listed in pg_dist_node with a different nodecluster value, and the servers have a corresponding citus.cluster_name in their postgresql.conf. That way, nodes know which other nodes in pg_dist_node belong to their cluster, and they will ignore others.
+
+**Read replica cluster**: In a read replica cluster, every node is a physical replica of a node in a primary Citus cluster. The read replica has a distinct citus.cluster_name value and the nodes in the read replica cluster should be added to pg_dist_node on the primary coordinator with the corresponding cluster name.
+
+**Client connections**: Connections made by the client to any of the nodes in the cluster. Each client connection is backed by a postgres process/backend, which we sometimes refer to as a client session.
+
+**Internal connections**: Connections to other nodes made by a client session. Each internal connection is backed by a process, which we sometimes refer to as an internal session. In the code, you can use IsCitusInternalBackend()
+
+**Maintenance daemon**: A background worker that is started in each database that has the Citus extension. It performs distributed deadlock detection, 2PC recovery, synchronizing node metadata after citus_update_node, resource cleanup, and other tasks.
+
+In the query planner, we use the following terminology:
+
+**Distributed query**: A query sent by the client that involves a Citus table and is therefore handled by the distributed query planner.
+
+**Shard query**: An internal query on shards (at most 1 shard group).
+
+**Intermediate result**: A temporary file that contains the result of a subquery or CTE, created as a result of a broadcast or repartition operation. Intermediate results are automatically cleaned up on transaction end or restart.
+
+## Principles
+
+Use cases:
+
+- Multi-tenant apps are the primary use case for Citus, which we can scale through distributing and co-locating by tenant ID, or through schema-based sharding. Citus is reasonably complete for this use case, but there are still SQL and operational improvements that can be made.
+- Real-time analytics is another popular use case due the combination of parallel distributed queries with indexes & in-database materialization (ETL). Improvement areas are automated time partitioning, better columnar storage (perf and update/delete, and incremental materialized views).
+- Citus works well for CRUD use cases, but would be far easier to use if we introduced a load balancer, DDL from any node (no explicit coordinator), and more performant by better use of connection pooling (e.g. outbound pgbouncers).
```

Comment:
```suggestion
- Citus works well for CRUD use cases, but would be far easier to use if we introduced a load balancer and DDL from any node (no explicit coordinator); and would be more performant by better use of connection pooling (e.g. outbound pgbouncers).
```

Reply:
I'll push a slight variation of this suggestion soon